maven/JarAssembler.kt

```diff
@@ -0,0 +1,68 @@
+import picocli.CommandLine
+import picocli.CommandLine.Command
+import picocli.CommandLine.Option
+import java.io.BufferedInputStream
+import java.io.BufferedOutputStream
+import java.io.File
+import java.io.FileOutputStream
+import java.util.*
+import java.util.concurrent.Callable
+import java.util.zip.ZipEntry
+import java.util.zip.ZipFile
+import java.util.zip.ZipOutputStream
+import kotlin.system.exitProcess
+
+
+@Command(name = "pom_generator", mixinStandardHelpOptions = true)
+class JarAssembler : Callable<Unit> {
+
+    @Option(names = ["--output"], required = true)
+    lateinit var output_file: File
+
+    @Option(names = ["--prefix"])
+    lateinit var prefix: String
+
+    @Option(names = ["--group-id"])
+    var groupId = ""
+
+    @Option(names = ["--artifact-id"])
+    var artifactId = ""
+
+    @Option(names = ["--pom-file"])
+    var pomFile: File? = null
+
+    @Option(names = ["--jars"], split = ";")
+    lateinit var jars: Array<File>
+
+    val entryNames = mutableSetOf<String>()
+
+    override fun call() {
```

Comment:
Can we merge `call` into `main`?

I want the main function to contain the logic of the code. Having a `main` function of which its only job is to execute `call` is not intuitive I think

Reply:
@lolski 

> Can we merge call into main?
> I want the main function to contain the logic of the code

I don't think we can actually do that; I'm getting this error when trying to do so:

```
error: class 'PomGeneratorOptions' is not abstract and does not implement abstract member public abstract fun call(): Unit! defined in java.util.concurrent.Callable
```

>Having a main function of which its only job is to execute call is not intuitive I think

That's something that `picocli`'s documentation actually suggests: you delegate all the actual works to the `@Command` classes - we just so happen to only have one command per file.

Reply:
got it